integration-test/src/main/java/com/sequenceiq/it/cloudbreak/dto/mock/endpoint/ClouderaManagerEndpoints.java

```diff
@@ -1,157 +1,201 @@
-import com.cloudera.api.swagger.model.ApiRole;
-import com.cloudera.api.swagger.model.ApiRoleList;
-import com.cloudera.api.swagger.model.ApiService;
-import com.cloudera.api.swagger.model.ApiServiceList;
 import com.cloudera.api.swagger.model.ApiUser2;
-import com.cloudera.api.swagger.model.ApiUser2List;
-import com.cloudera.api.swagger.model.ApiVersionInfo;
+import com.sequenceiq.it.cloudbreak.dto.CloudbreakTestDto;
 import com.sequenceiq.it.cloudbreak.dto.mock.SparkUri;
-import com.sequenceiq.it.cloudbreak.dto.mock.answer.AnswerWithoutRequest;
-import com.sequenceiq.it.cloudbreak.dto.mock.answer.ClouderaManagerPreparedRequestAnswer;
-import com.sequenceiq.it.cloudbreak.dto.mock.answer.StringRequestAnswer;
+import com.sequenceiq.it.cloudbreak.dto.mock.answer.DefaultResponseConfigure;
+import com.sequenceiq.it.cloudbreak.mock.ExecuteQueryToMockInfrastructure;
 
-public final class ClouderaManagerEndpoints {
+public final class ClouderaManagerEndpoints<T extends CloudbreakTestDto> {
     public static final String API_ROOT = "/api/v31";
 
-    private ClouderaManagerEndpoints() {
+    private T testDto;
+
+    private ExecuteQueryToMockInfrastructure executeQueryToMockInfrastructure;
+
+    public ClouderaManagerEndpoints(T testDto, ExecuteQueryToMockInfrastructure executeQueryToMockInfrastructure) {
+        this.testDto = testDto;
+        this.executeQueryToMockInfrastructure = executeQueryToMockInfrastructure;
+    }
+
+    public Users<T> users() {
+        return (Users<T>) EndpointProxyFactory.create(Users.class, testDto, executeQueryToMockInfrastructure);
+    }
+
+    public Admin<T> usersAdmin() {
+        return (Admin<T>) EndpointProxyFactory.create(Admin.class, testDto, executeQueryToMockInfrastructure);
+    }
+
+    public ClusterHosts<T> clusterHosts() {
+        return (ClusterHosts<T>) EndpointProxyFactory.create(ClusterHosts.class, testDto, executeQueryToMockInfrastructure);
+    }
+
+    public ClusterCommands.DeployConfig<T> clusterCommandsDeployConfig() {
+        return (ClusterCommands.DeployConfig<T>) EndpointProxyFactory.create(ClusterCommands.DeployConfig.class, testDto, executeQueryToMockInfrastructure);
+    }
+
+    public ClusterHostTemplates.CommandsApplyHostTemplate<T> commandsApplyHostTemplate() {
+        return (ClusterHostTemplates.CommandsApplyHostTemplate<T>)
+                EndpointProxyFactory.create(ClusterHostTemplates.CommandsApplyHostTemplate.class, testDto, executeQueryToMockInfrastructure);
+    }
+
+    public Commands<T> commands() {
+        return (Commands<T>) EndpointProxyFactory.create(Commands.class, testDto, executeQueryToMockInfrastructure);
+    }
+
+    public ClouderaManagerHosts<T> clouderaManagerHosts() {
+        return (ClouderaManagerHosts<T>) EndpointProxyFactory.create(ClouderaManagerHosts.class, testDto, executeQueryToMockInfrastructure);
     }
 
     @SparkUri(url = API_ROOT + "/cm/importClusterTemplate", requestType = ApiClusterTemplate.class)
```

Comment:
If sparkservice eliminated we might consider to rename this annotation,or even yield this from a swagger doc (if I recall correctly mocks are generated based on swagger)